src/jti_background.js

```diff
@@ -213,6 +254,80 @@ function addTemplate (templateName, issueTypeField, projectsField, text, callbac
     });
 }
 
+function addDomain (domainName, callback) {
+    chrome.storage.sync.get(StorageID, function (data) {
+        var domainJSON = {};
+
+        if (data[StorageID]) {
+            domainJSON = data[StorageID];
+        }
+
+        var domainID = getNextID(domainJSON.options.domains);
+        var newDomain = {
+            'id': domainID,
+            'name': domainName
+        };
+
+        validateDomain(domainName, function (message) {
+            if (message) {
+                callback(false, message);
+            } else {
+                domainJSON.options.domains[domainID] = newDomain;
+                // Refresh existing pages with this URL.
+                matchRegexToJsRegex(domainName);
+                chrome.tabs.query({windowId: chrome.windows.WINDOW_ID_CURRENT}, function (tabs) {
+                    $.each(tabs, function (tabIndex, tab) {
+                        var chromeRegex = new RegExp('chrome://extensions');
```

Comment:
Why are we testing the extension page?

Reply:
👍

Reply:
Actually I think it would make more sense to push this check to the validation of the domain being inserted. i.e don't allow a user to add `chrome://extension`
That way we only need to check for it in a single place.

Reply:
I agree with you, that's a much better way to do it. 

The difficulty is that we reload domains based on regex (atlassian.net reloads all pages with atlassian.net in the url). That means that some strings that should be allowed will also trigger a reload of chrome://extensions. We'd have to exclude _all_ substrings of chrome://extensions during validation, which we don't want to do.

Reply:
👍  Ok we can leave this. Maybe just add a comment explaining why.